samples/net/updatehub/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ if(CONFIG_UPDATEHUB_DTLS)
             privkey.der
             )
         generate_inc_file_for_target(
-            ${ZEPHYR_CURRENT_LIBRARY}
```

Comment:
Before boilerplate.cmake returns it creates a zephyr_library called '${IMAGE}app' this is by-design and intentional.

This has ZEPHYR_CURRENT_LIBRARY evaluate to 'app'.

Referencing ZEPHYR_CURRENT_LIBRARY instead of app is completely valid and a supported part of the API.

That being said, it is more straightforward and less convoluted to just use 'app', so the patch is fine,
but the commit message should be modified according to the explanation above.

Reply:
Ah, didn't read this commit message:

https://github.com/zephyrproject-rtos/zephyr/pull/16737/commits/7b1de25b9deff2967e325f3f9a66f81bf59f6ce5

you want to make it internal, hmm ... I need to think about that.

Github re-orders the commits in the PR, so I could have read this in the wrong order.